### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'JsonRpcProvider')

### DISCUSSION THREAD

**Discussion by endresol** [NomicFoundation/hardhat#3650] [Troubleshooting] [answered]
Hello. 

I am getting this error when trying to deploy a contract with parameters and I can not understand why. 
According to the dump, it is from this line:
      at Object.<anonymous> (node_modules\@nomiclabs\hardhat-ethers\src\internal\ethers-provider-wrapper.ts:4:61)

I expect it is something I do, but the error message makes it very hard to understand what...
Any tips on how to find out what is wrong?

**Comment by endresol** [comment, marked as answer]
OK, I have done some research and the error is related to ethers. When I downgrade to ^5.4.7 (from ^6.0.2) it works.

**Reply by endresol** [reply]
Hardhat 6.1.2 does not support ethers.js 6.0.2. Downgrade to 5.4.7 until support is upgraded. 
Follow this issue for more info: https://github.com/NomicFoundation/hardhat/issues/3639

**Reply by fvictorio** [reply]
Glad you figured out and thanks for self-answering for future readers :slightly_smiling_face: Adding support for the new version of Ethers is one of our top priorities right now.

**Comment by k06a** [comment]
Wish `@nomiclabs/ethers-provider-wrapper` to support `ethers@6`
